### PR TITLE
Add Master Hexagonal Architecture part 2 and Rust Ownership Explained

### PR DESCRIPTION
**Pull request by AngusGMorrison**
Adds two new walkthroughs:

- The second part of _Master Hexagonal Architecture in Rust_. Part two of this series, _Separation of Concerns, Rust-Style_ walks readers through the process of moving from a big ball of mud towards an application loosely coupled by traits.
  - It might not be obvious from the link title that this is part 2 of the series. Perhaps [Master Hexagonal Architecture in Rust (part 2): Separation of Concerns, Rust-Style] would be an appropriate edit?
- A standalone walkthrough, _Rust Ownership Explained: Merging Linked Lists_. This is pitched at a less advanced audience and aims to build intuition with ownership, references and `std::mem::swap` through a linked list coding challenge with a surprising twist.

### CODE REVIEW

File: draft/2024-06-19-this-week-in-rust.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ and just ask the editors to select the category.
 ### Observations/Thoughts
 
 ### Rust Walkthroughs
+- [series] [Master Hexagonal Architecture in Rust: Separation of Concerns, Rust-Style](https://www.howtocodeit.com/articles/master-hexagonal-architecture-rust#separation-of-concerns-rust-style)
+- [Rust Ownership Explained: Merging Linked Lists](https://www.howtocodeit.com/articles/rust-ownership-explained-linked-lists)
```
Comment (mariannegoldin) [inline draft/2024-06-19-this-week-in-rust.md:42]: Editing to add series annotation (thanks for the suggestion) and to fix formatting:
```suggestion
* [series] [Master Hexagonal Architecture in Rust (part 2): Separation of Concerns, Rust-Style](https://www.howtocodeit.com/articles/master-hexagonal-architecture-rust#separation-of-concerns-rust-style)
* [Rust Ownership Explained: Merging Linked Lists](https://www.howtocodeit.com/articles/rust-ownership-explained-linked-lists)
```

**Review by mariannegoldin — APPROVED** [review]
Thanks for the submission, we love these walkthroughs. For some reason, I am unable to commit my edits. I will go ahead and approve and merge this PR as-as and make the changes in a separate PR. As a heads up, the changes I am going to make are:
* Replace "-" with "*" bullets
* Add in "part 2" as you suggested - yes, this is the preferred title scheme for a series

**Comment by mariannegoldin** [comment]
Going to make this update in a patch PR #5637 as I was not able to edit this one. Closing this PR only for that reason.

**Comment by AngusGMorrison** [comment]
> Thanks for the submission, we love these walkthroughs. For some reason, I am unable to commit my edits. I will go ahead and approve and merge this PR as-as and make the changes in a separate PR. As a heads up, the changes I am going to make are:
> 
> * Replace "-" with "*" bullets
> * Add in "part 2" as you suggested - yes, this is the preferred title scheme for a series

Wonderful, thanks. Apologies for the formatting!

**Pull request closed without merging**